### DISCUSSION TITLE
Should exceptions be captured by Sentry-Rails?

### DISCUSSION THREAD

**Discussion by renchap** [janko/rodauth-rails#171] [Q&A]
I just noticed I had a bug for an edge-case in a custom Rodauth feature, raising the following exception in development:

```
lib/rodauth_verify_account_forever_grace.rb:46:in `before_change_login_route'
app/lib/rodauth_app.rb:392:in `block (2 levels) in <class:RodauthApp>'
app/lib/rodauth_app.rb:379:in `block in <class:RodauthApp>'
Completed 500 Internal Server Error in 51ms (Views: 0.1ms | ActiveRecord: 7.9ms | Allocations: 7815)
```

But this exception has not been reported to Sentry in production, even if I know this happened to some users.

Is this expected? Do you have an idea why it was not properly catched by Sentry?

**Comment by janko** [comment]
Hmm, I have no idea why that hasn't happened, I agree exceptions absolutely should be caught by exception monitoring tools such as Sentry. I do recall sentry-rails gem having a Rack middleware, but exceptions from the Rodauth app should be bubbling up the middleware stack. Are you able to reproduce this behaviour in a fresh Rails application?
